pkgs/tools/archivers/rar/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, fetchurl, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "rar";
+  version = "6.0.2";
+
+  src = fetchurl {
+    url = "https://www.rarlab.com/rar/rarlinux-x64-${version}.tar.gz";
```

Comment:
FYI, there are also:
- x86 builds (I don't think x86 is supported in NixOS though)
- macOS
- FreeBSD

Reply:
[i686-linux is tier 3](https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md#tier-3-1). If you add --argstr system i686-linux, you should be able to test it on x86_64 Linux. (on flakes, use `legacyPackages.i686-linux`) Please give it a try.